static/app/components/tagsTable.tsx

```diff
@@ -20,83 +20,90 @@ type Props = {
+const getErrorMessage = (error: MetaError) => {
+  if (Array.isArray(error)) {
+    if (error[1]?.reason) {
+      return formatErrorKind(error[1].reason);
     }
-    return formatErrorKind(error);
-  };
+    return formatErrorKind(error[0]);
+  }
+  return formatErrorKind(error);
+};
 
-  const getTooltipTitle = (errors: Array<MetaError>) => {
-    return <TooltipTitle>{getErrorMessage(errors[0])}</TooltipTitle>;
-  };
+const getTooltipTitle = (errors: Array<MetaError>) => {
+  return <TooltipTitle>{getErrorMessage(errors[0])}</TooltipTitle>;
+};
+
+const TagsTable = ({event, query, generateUrl, title = t('Tag Details')}: Props) => {
+  const eventWithMeta = withMeta(event) as Event;
+  const tags = eventWithMeta.tags;
 
   return (
     <StyledTagsTable>
       <SectionHeading>{title}</SectionHeading>
       <KeyValueTable>
-        {tags.map(tag => {
-          const tagInQuery = query.includes(`${tag.key}:`);
-          const target = tagInQuery ? undefined : generateUrl(tag);
-          const keyMetaData = getMeta(tag, 'key');
-          const valueMetaData = getMeta(tag, 'value');
-
-          const renderTagValue = () => {
-            switch (tag.key) {
-              case 'release':
-                return <Version version={tag.value} anchor={false} withPackage />;
-              default:
-                return tag.value;
-            }
-          };
-          return (
-            <KeyValueTableRow
-              key={tag.key}
-              keyName={
-                keyMetaData?.err?.length ? (
-                  <Tooltip title={getTooltipTitle(keyMetaData.err)}>
-                    <i>{`<${t('invalid')}>`}</i>
-                  </Tooltip>
-                ) : (
-                  <StyledTooltip title={tag.key}>{tag.key}</StyledTooltip>
-                )
-              }
-              value={
-                valueMetaData?.err?.length ? (
-                  <Tooltip title={getTooltipTitle(valueMetaData.err)}>
-                    <i>{`<${t('invalid')}>`}</i>
-                  </Tooltip>
-                ) : keyMetaData?.err?.length ? (
-                  <span>{renderTagValue()}</span>
-                ) : tagInQuery ? (
-                  <Tooltip title={t('This tag is in the current filter conditions')}>
-                    <span>{renderTagValue()}</span>
-                  </Tooltip>
-                ) : (
-                  <StyledTooltip title={renderTagValue()}>
-                    <Link to={target || ''}>{renderTagValue()}</Link>
-                  </StyledTooltip>
-                )
-              }
-            />
-          );
-        })}
+        {tags.map(tag => (
+          <TagsTableRow key={tag.key} tag={tag} query={query} generateUrl={generateUrl} />
+        ))}
       </KeyValueTable>
     </StyledTagsTable>
   );
 };
 
+interface TagsTableRowsProps extends Omit<Props, 'event' | 'title'> {
+  tag: EventTag;
+}
+
+export const TagsTableRow = ({tag, query, generateUrl}: TagsTableRowsProps) => {
```

Comment:
Should this live in its own file if it's going to be used outside of `<TagsTable>`?

Reply:
yeah, it's enough of it's own thing to be separate.

I'd float that it's a reasonable TSC rule that each file should have one `interface Props {...};` definition and none `interface MyFooProps {...};` definitions. Just make a 2nd file.